gtfs/__main__.py

```diff
@@ -1,25 +1,83 @@
 #!/usr/bin/env python
 """Command line interface for fetching GTFS."""
-import argparse
-import getpass
 import logging
-import sys
 
-from prettytable import PrettyTable
+import typer
+from chalky import chain
+from halo import Halo
+from prettytable.colortable import ColorTable, Themes
+from typing_extensions import Annotated
 
-from gtfs.utils.FeedSource import FeedSource
-import feed_sources
-# import all the available feed sources
-# pylint: disable=I0011,wildcard-import
-from feed_sources import *
+from .feed_source import FeedSource
+from .feed_sources import __all__ as feed_sources
+from .utils.constants import Predicate
+from .utils.geom import bbox_contains_bbox, bbox_intersects_bbox
 
 logging.basicConfig()
 LOG = logging.getLogger()
 LOG.setLevel(logging.INFO)
+app = typer.Typer()
 
-def fetch_all(sources=None):
-    """Fetch from all FeedSources in the feed_sources directory.
 
+def check_bbox(bbox: str):
+    if "," not in bbox or len(bbox.split(",")) != 4:
+        raise typer.BadParameter(
+            chain.bright_red
+            | "Please pass bbox as a string separated by commas like this: xmin,ymin,xmax,ymax"
+        )
+    for coord in bbox.split(","):
+        try:
+            float(coord)
+        except ValueError:
+            raise typer.BadParameter(chain.bright_red | "Please pass only numbers as bbox values!")
+    return bbox
+
+
+@app.command()
+def list_feeds(
+    bbox: Annotated[
+        str,
+        typer.Argument(
+            help="pass value as a string separated by commas like this: xmin,ymin,xmax,ymax ",
+            callback=check_bbox,
+        ),
+    ],
+    predicate: Annotated[
+        Predicate,
+        typer.Option(
+            "--predicate",
+            "-p",
+            help="the gtfs feed should intersect or should be contained inside the user's bbox",
+        ),
+    ] = Predicate.intersects,
+):
+    """Filter feeds spatially based on bounding box."""
+    spinner = Halo(text="Filtering...", text_color="cyan", spinner="dots")
+    spinner.start()
+    user_bbox = [float(coord) for coord in bbox.split(",")]
+    ptable = ColorTable(["Feed Source", "Transit URL", "Bounding Box"], theme=Themes.OCEAN)
+    for src in feed_sources:
+        feed_bbox = src.bbox
+        if predicate == "contains":
+            if bbox_contains_bbox(feed_bbox, user_bbox):
+                row = [src, chain.bright_yellow | src.url, feed_bbox]
+                ptable.add_row(row)
+        else:
+            if bbox_intersects_bbox(feed_bbox, user_bbox):
+                row = [src, chain.bright_yellow | src.url, feed_bbox]
+                ptable.add_row(row)
```

Comment:
you repeat the same thing twice

```suggestion
        if predicate == "contains" 
            if not bbox_contains_bbox(feed_bbox, user_bbox):
                continue
        elif predicate == "intersects":
            if not bbox_intersects_bbox(feed_bbox, user_bbox):
                continue
   
        row = [src, chain.bright_yellow | src.url, feed_bbox]
        ptable.add_row(row)
```